src/Logary.Tests/Metrics.fs

```diff
@@ -0,0 +1,168 @@
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> registry.registerMetricWithNoLables
+    gauge.set 200.
+    gauge.inc 100.
+    gauge.inc -50.
+    gauge.dec 50.
+    gauge.dec -50.
+    let _, MetricInfo.Gauge gaugeInfo = gauge.explore ()
+    gaugeInfo.gaugeValue |> Expect.equal "gauge value should be 250" 250.
+
+  testCase "gauge with histogram" <| fun () ->
+    let testBuckets = [| 5.; 10.; 50.; 100.; |]
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> GaugeConf.withHistogram testBuckets |> registry.registerMetricWithNoLables
+
+
+    gauge.set 200.
+    gauge.dec 140. // 60
+    gauge.dec 15.  // 45
+    gauge.dec 5.   // 40
+    gauge.dec 15.  // 25
+    gauge.dec 10.  // 15
+    gauge.dec 8.   // 7
+    gauge.inc 18.  // 25
+
+    let metricInfos = registry.getMetrictInfos ()
+    let _, metricDetails = metricInfos |> Seq.find (fun (basicInfo , _) -> basicInfo.name = gaugeConf.basicInfo.name)
+    metricDetails |> Expect.sequenceEqual "should have one gauge detail" [ MetricInfo.Gauge { labels= Map.empty ; gaugeValue = 25. } ]
+
+    let _, metricDetails = metricInfos |> Seq.find (fun (basicInfo , _) -> basicInfo.name = sprintf "%s_histogram" gaugeConf.basicInfo.name)
+    let bucketsInfo = [(5.,0.); (10.,1.); (50., 5.); (100., 1.); (Double.PositiveInfinity, 1.)] |> Map.ofSeq
+    metricDetails |> Expect.sequenceEqual "should have histogram detail" [ MetricInfo.Histogram { labels= Map.empty ; bucketsInfo = bucketsInfo; sumInfo = float(200+60+45+40+25+15+7+25) } ]
+
+    let registry = new MetricRegistry()
+    let histogramConf = HistogramConf.create("some histogram", "some histogram") |> HistogramConf.buckets testBuckets
+    let gauge = gaugeConf |> GaugeConf.enableHistogram histogramConf |> registry.registerMetricWithNoLables
+
+    let metricInfos = registry.getMetrictInfos ()
+    let _, metricDetails = metricInfos |> Seq.find (fun (basicInfo , _) -> basicInfo.name = gaugeConf.basicInfo.name)
+    metricDetails |> Expect.sequenceEqual "should have one empty gauge detail" [ MetricInfo.Gauge { labels= Map.empty ; gaugeValue = 0. } ]
+
+    let _, metricDetails = metricInfos |> Seq.find (fun (basicInfo , _) -> basicInfo.name = histogramConf.basicInfo.name)
+    let bucketsInfo = testBuckets |> Seq.map (fun i -> (i, 0.)) |> Map.ofSeq |> Map.add Double.PositiveInfinity 0.
+    metricDetails |> Expect.sequenceEqual "should have empty histogram detail" [ MetricInfo.Histogram { labels= Map.empty ; bucketsInfo = bucketsInfo; sumInfo = 0. } ]
+
+
+  testCase "histogram with lable" <| fun () ->
+    let testBuckets = [| 5.; 10.; 50.; 100.; |]
+    let registry = new MetricRegistry()
+
+    let histogramConf = HistogramConf.create {name = "some histogram"; description = "some histogram"; labelNames = [| "queue name" |]} |> HistogramConf.buckets testBuckets
+    let histogram = histogramConf |> registry.registerMetric
+
+    (histogram.labels [| "queue a" |]).observe 15.
```

Comment:
`Time` here can be express as `Span` in logary. 
```f#
    let lgb = logm.getLogger "logger.b"
    use _ = lgb |> Span.create |> Span.setMessage (eventX "some child span") |> Span.start
    // do actual work.
``` 
after span finished(disposed), it will have a timespan info which will be used for gauges.

Reply:
True